### ISSUE TITLE
mouse navigation projects

### ISSUE THREAD

**Issue by molleweide** [kmonad/kmonad#322]
# move mouse with keys

## macos

Easy to modify lua script. this is the only one i have used myself and it works well but it is not efficient enough yet.

https://github.com/tweekmonster/hammerspoon-vimouse

## linux

Mouse utility for X11 written in haskell:

https://github.com/unclechu/place-cursor-at

There are also these written in C:

https://github.com/jordansissel/keynav
https://github.com/SalmonChris/KeyboardMouse
https://github.com/Ph0enixKM/Mousepad

## windows

I haven't been able to find anything on github but searching seems to give many results..
https://duckduckgo.com/?q=windows+move+mouse+with+keyboard&ia=web

**Comment by SeerLite** [comment]
[keynav](https://www.semicomplete.com/projects/keynav/) is an interesting one that uses splits to quickly move the mouse to a specific place on the screen. Also allows mapping to different mouse buttons.

I remember using it one time and it worked well enough, but never got used to it. Maybe with some practice one can get good at it.

**Comment by molleweide** [comment]
That also seems pretty cool!

**Comment by molleweide** [comment]
I have to look i to it later. My personal goal is to create some kind of api for kmo so that you can taylor your own pointer.

**Comment by Ultrahalf** [comment]
```
  ;; MOUSE KEYS
  mtog  #((cmd-button "setxkbmap -option keypad:pointerkeys") P1000 S-nlck)
  mml kp4
  mmd kp2
  mmu kp8
  mmr kp6
  mwl (cmd-button "xdotool click 6")
  mwd (cmd-button "xdotool click 5")
  mwu (cmd-button "xdotool click 4")
  mwr (cmd-button "xdotool click 7")
  mb1 #(kp/ kp5)
  mb2 #(kp- kp5)
  mb3 #(kp* kp5)
```
got this from miryoku's kmonad config
work pretty good but for some reason the wheel keys wont repeat in my config

**Comment by Ultrahalf** [comment]
https://github.com/rvaiya/warpd
if anyone wants to try

**Comment by david-janssen** [comment]
Sorry to burst your bubbles, but anything that depends on X11 is never going to get merged into KMonad. Although I would love to support interesting interface hacks, a core design decision of KMonad was to keep it as close to kernel as possible. That way we can also easily support Mac, Windows, and I'm guessing Wayland.

However, there is nothing to stop you from developing your own mouse-placement software and hooking it up to kmonad through some keybindings.

Going to close this, because sometimes it's better to rip the bandaid of quickly. Sorry my dudes.

**Comment by slotThe** [comment]
I think this was one of these "not really an issue but posing as an issue anyways for ease of discoverability—but really, we should use GitHub Discussions (or the wiki) for this"-type of issue :)

**Comment by molleweide** [comment]
Yeah i just havent gotten used to using discissions so that's why i posted here initially

**Comment by molleweide** [comment]
> Sorry to burst your bubbles, but anything that depends on X11 is never going to get merged into KMonad. Although I would love to support interesting interface hacks, a core design decision of KMonad was to keep it as close to kernel as possible. That way we can also easily support Mac, Windows, and I'm guessing Wayland.
> 
> 
> 
> However, there is nothing to stop you from developing your own mouse-placement software and hooking it up to kmonad through some keybindings.
> 
> 
> 
> Going to close this, because sometimes it's better to rip the bandaid of quickly. Sorry my dudes.

I agree with everything

**Comment by leosolid** [comment]
+1 to enabling discussions to continue discussing this topic

**Comment by david-janssen** [comment]
> +1 to enabling discussions to continue discussing this topic

If they weren't enabled already they should be available now. Viel spass!
